erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py

```diff
@@ -1,67 +1,105 @@
-# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# Copyright (c) 2023, Frappe Technologies Pvt. Ltd. and contributors
 # For license information, please see license.txt
 
-from datetime import date
-
 import frappe
+from frappe import qb
 from frappe.model.document import Document
-
-from erpnext.utilities.bulk_transaction import task, update_logger
+from frappe.query_builder.functions import Count
+from frappe.utils import cint
+from pypika import Order
 
 
 class BulkTransactionLog(Document):
-	pass
-
-
-@frappe.whitelist()
-def retry_failing_transaction(log_date=None):
-	if not log_date:
-		log_date = str(date.today())
-	btp = frappe.qb.DocType("Bulk Transaction Log Detail")
-	data = (
-		frappe.qb.from_(btp)
-		.select(btp.transaction_name, btp.from_doctype, btp.to_doctype)
-		.distinct()
-		.where(btp.retried != 1)
-		.where(btp.transaction_status == "Failed")
-		.where(btp.date == log_date)
-	).run(as_dict=True)
-
-	if data:
-		if len(data) > 10:
-			frappe.enqueue(job, queue="long", job_name="bulk_retry", data=data, log_date=log_date)
-		else:
-			job(data, log_date)
-	else:
-		return "No Failed Records"
-
-
-def job(data, log_date):
-	for d in data:
-		failed = []
-		try:
-			frappe.db.savepoint("before_creation_of_record")
-			task(d.transaction_name, d.from_doctype, d.to_doctype)
-		except Exception as e:
-			frappe.db.rollback(save_point="before_creation_of_record")
-			failed.append(e)
-			update_logger(
-				d.transaction_name,
-				e,
-				d.from_doctype,
-				d.to_doctype,
-				status="Failed",
-				log_date=log_date,
-				restarted=1,
-			)
+	def db_insert(self, *args, **kwargs):
+		pass
+
+	def load_from_db(self):
+		log_detail = qb.DocType("Bulk Transaction Log Detail")
+		succeeded_logs = (
```

Comment:
This type of query might give an index error if there are no `Bulk Transaction Log Detail` and the result is `None`.

Reply:
`DoesNotExistError` exception will be raised, if an invalid date or if there are no logs on that date.
<img width="1552" alt="Screenshot 2023-11-13 at 10 43 20 AM" src="https://github.com/frappe/erpnext/assets/3272205/0d894f8b-13ac-41c8-ac68-a5980b8e334c">